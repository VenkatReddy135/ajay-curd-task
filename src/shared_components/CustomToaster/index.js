/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";
import { ToastContainer, Toast } from "react-bootstrap";

/**
 *
 * @param {object} props all props
 *@function CustomToaster
 * @returns {React.createElement} CustomToaster
 */
export default function CustomToaster(props) {
  return (
    <ToastContainer position="top-end" className="p-3">
      <Toast
        onClose={props.handleClose}
        show={props.showToast}
        bg="success"
        autohide
      >
        <Toast.Body className="text-white">
          {props.successMessage || null}
        </Toast.Body>
      </Toast>
    </ToastContainer>
  );
}

CustomToaster.propTypes = {
  showToast: PropTypes.bool.isRequired,
  handleClose: PropTypes.func,
  successMessage: PropTypes.string,
};

CustomToaster.defaultProps = {
  showToast: false,
};
