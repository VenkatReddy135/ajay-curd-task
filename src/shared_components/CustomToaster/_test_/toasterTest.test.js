import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CustomToaster from "../index";
import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<CustomToaster />);
});
const props = {
  showToast: false,
};
describe("Toaster Component Testing", () => {
  it("renders component", () => {
    const tree = renderer.create(<CustomToaster {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CustomToaster {...props} />, div);
  });
  it("should have Toast", () => {
    console.log(wrapper);
    expect(wrapper.find("Toast")).toHaveLength(1);
  });

  afterEach(() => {
    cleanup();
  });
});
