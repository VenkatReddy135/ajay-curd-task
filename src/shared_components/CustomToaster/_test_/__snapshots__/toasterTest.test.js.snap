// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster Component Testing renders component 1`] = `
<div
  className="toast-container position-absolute top-0 end-0 p-3"
/>
`;
